metron-deployment/packaging/ambari/README.md

```diff
@@ -0,0 +1,315 @@
+         )
+    # Do stuff
+```
+This will create a file on the Ambari agent machine's file system, `/etc/sysconfig/metron`, with the content of `metron.j2`, after replacing `{{metron_log_dir}}` with the value of the property (`/var/log/metron`)
+```
+...
+METRON_LOG_DIR="/var/log/metron"
+...
+```
+
+
+
+## How to identify errors in MPack changes
+Typically, any errors are thrown at one of two times:
+
+1. Attempting to install Metron as a whole.  These are typically service level definition errors, not property errors. Logs will often be found in `/var/log/ambari-server/ambari-server.log`.  Often the log will indicate it was unable to load a particular service or subservice with some more detail.
+2. Running the actual functionality.  These typically tend to cause errors in the UI at runtime.  Logs are usually found in `/var/log/ambari-agent/ambari-agent.log`.
+
+Unfortunately, because errors tend to occur at runtime, it's often necessary to add things like logging statements, or even just throw errors to print out in the Ambari UI.
+
+The primary solution to these is to look in the logs for exceptions, see what's going wrong (Property doesn't exist?  Malformed file couldn't be loaded?), and adjust appropriately.
+
+## Testing changes without cycling Vagrant build
+To avoid spinning down and spinning back up Vagrant, we'll want to instead modify the files Ambari uses directly.
+
+This assumes the installation went through, and we're just working on getting our particular feature / adjustment to work properly.
+
+Ambari stores the Python files from the service in a couple places.  We'll want to update the files, then have Ambari pick up the updated versions and use them as the new basis.  A reinstall of Metron is unnecessary for this type of testing.
+
+Specifically, the server files live in
+```
+/var/lib/ambari-server/resources/mpacks/metron-ambari.mpack-0.4.0.0/common-services
+/var/lib/ambari-agent/cache/common-services
+```
+
+e.g. enrichment_commands.py can be found in:
+
+```
+/var/lib/ambari-server/resources/mpacks/metron-ambari.mpack-0.4.0.0/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+/var/lib/ambari-agent/cache/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+```
+
+A `find` command can also be useful in quickly locating the exact location of a file, e.g.
+```
+[root@node1 ~]# find /var/lib/ -name enrichment_commands.py
+/var/lib/ambari-server/resources/mpacks/metron-ambari.mpack-0.4.0.0/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+/var/lib/ambari-agent/cache/common-services/METRON/0.4.0/package/scripts/enrichment_commands.py
+```
+
+The steps to update are:
+
+1. Stop Metron through Ambari.  If the property in question is used by the topologies, we'll want them stopped, so they can be restarted with the new property.
+  * This can sometimes be skipped if the change doesn't affect the topologies themselves, but is a case by case choice.
```

Comment:
The formatting looks strange here when I view it on github.  Not a big deal though.